From 636d07f8a97beeb4c63137d755133fc2bcfad093 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Fri, 6 Oct 2023 10:22:51 +0200
Subject: [PATCH] Revert "Deprecation of com.webos.service.power"

This reverts commit 874ad6b9cd9ba8cdae0a548574524caaf3ad9227.

Since we still use these calls on LuneOS and they cause quite some logspam.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/pwrevents/shutdown.c    | 22 +++++++++++-----------
 src/pwrevents/suspend_ipc.c | 33 ++++++++++++++++-----------------
 2 files changed, 27 insertions(+), 28 deletions(-)

diff --git a/src/pwrevents/shutdown.c b/src/pwrevents/shutdown.c
index 86617aa..a5f04f9 100644
--- a/src/pwrevents/shutdown.c
+++ b/src/pwrevents/shutdown.c
@@ -1,4 +1,4 @@
-// Copyright (c) 2011-2023 LG Electronics, Inc.
+// Copyright (c) 2011-2021 LG Electronics, Inc.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -1210,26 +1210,26 @@ cleanup:
 
 LSMethod shutdown_methods[] =
 {
-    { "initiate", initiateShutdown,LUNA_METHOD_FLAG_DEPRECATED },
+    { "initiate", initiateShutdown },
 
-    { "shutdownApplicationsRegister", shutdownApplicationsRegister,LUNA_METHOD_FLAG_DEPRECATED },
-    { "shutdownApplicationsAck", shutdownApplicationsAck,LUNA_METHOD_FLAG_DEPRECATED },
+    { "shutdownApplicationsRegister", shutdownApplicationsRegister },
+    { "shutdownApplicationsAck", shutdownApplicationsAck },
 
-    { "shutdownServicesRegister", shutdownServicesRegister,LUNA_METHOD_FLAG_DEPRECATED },
-    { "shutdownServicesAck", shutdownServicesAck,LUNA_METHOD_FLAG_DEPRECATED },
+    { "shutdownServicesRegister", shutdownServicesRegister },
+    { "shutdownServicesAck", shutdownServicesAck },
 
-    { "TESTresetShutdownState", TESTresetShutdownState,LUNA_METHOD_FLAG_DEPRECATED },
+    { "TESTresetShutdownState", TESTresetShutdownState },
 
-    { "machineOff", machineOff,LUNA_METHOD_FLAG_DEPRECATED },
-    { "machineReboot", machineReboot,LUNA_METHOD_FLAG_DEPRECATED },
+    { "machineOff", machineOff },
+    { "machineReboot", machineReboot },
 
     { },
 };
 
 LSSignal shutdown_signals[] =
 {
-    { "shutdownApplications",LUNA_METHOD_FLAG_DEPRECATED },
-    { "shutdownServices",LUNA_METHOD_FLAG_DEPRECATED },
+    { "shutdownApplications" },
+    { "shutdownServices" },
     { },
 };
 
diff --git a/src/pwrevents/suspend_ipc.c b/src/pwrevents/suspend_ipc.c
index f119515..cbe9dad 100644
--- a/src/pwrevents/suspend_ipc.c
+++ b/src/pwrevents/suspend_ipc.c
@@ -1,4 +1,4 @@
-// Copyright (c) 2011-2023 LG Electronics, Inc.
+// Copyright (c) 2011-2021 LG Electronics, Inc.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -831,20 +831,19 @@ LSMethod com_palm_suspend_methods[] =
 
     /* suspend methods*/
 
+    { "suspendRequestRegister", suspendRequestRegister },
+    { "prepareSuspendRegister", prepareSuspendRegister },
+    { "suspendRequestAck", suspendRequestAck },
+    { "prepareSuspendAck", prepareSuspendAck },
+    { "forceSuspend", forceSuspendCallback },
+    { "identify", identifyCallback },
+    { "clientCancelByName", clientCancelByName },
+    { "resume", resumeCallback },
 
-    { "suspendRequestRegister", suspendRequestRegister,LUNA_METHOD_FLAG_DEPRECATED },
-    { "prepareSuspendRegister", prepareSuspendRegister,LUNA_METHOD_FLAG_DEPRECATED },
-    { "suspendRequestAck", suspendRequestAck,LUNA_METHOD_FLAG_DEPRECATED },
-    { "prepareSuspendAck", prepareSuspendAck,LUNA_METHOD_FLAG_DEPRECATED },
-    { "forceSuspend", forceSuspendCallback,LUNA_METHOD_FLAG_DEPRECATED },
-    { "identify", identifyCallback,LUNA_METHOD_FLAG_DEPRECATED },
-    { "clientCancelByName", clientCancelByName,LUNA_METHOD_FLAG_DEPRECATED },
-    { "resume", resumeCallback,LUNA_METHOD_FLAG_DEPRECATED },
+    { "activityStart", activityStartCallback },
+    { "activityEnd", activityEndCallback },
 
-    { "activityStart", activityStartCallback,LUNA_METHOD_FLAG_DEPRECATED },
-    { "activityEnd", activityEndCallback, LUNA_METHOD_FLAG_DEPRECATED },
-
-    { "TESTSuspend", TESTSuspendCallback,LUNA_METHOD_FLAG_DEPRECATED },
+    { "TESTSuspend", TESTSuspendCallback },
 
     { },
 };
@@ -854,10 +853,10 @@ LSSignal com_palm_suspend_signals[] =
 
     /* Suspend signals */
 
-    { "suspendRequest",LUNA_METHOD_FLAG_DEPRECATED },
-    { "prepareSuspend",LUNA_METHOD_FLAG_DEPRECATED },
-    { "suspended",LUNA_METHOD_FLAG_DEPRECATED },
-    { "resume",LUNA_METHOD_FLAG_DEPRECATED },
+    { "suspendRequest" },
+    { "prepareSuspend" },
+    { "suspended" },
+    { "resume" },
 
     { },
 };
